src/res/enkf/ert_run_context.py

```diff
@@ -127,70 +46,57 @@ def ensemble_smoother_update(
         target_fs,
     ):
         return cls(
-            run_type=EnkfRunType.SMOOTHER_UPDATE,
-            mask=[],
-            sim_fs=sim_fs,
-            target_fs=target_fs,
-            paths=[],
-            jobnames=[],
+            EnkfRunType.SMOOTHER_UPDATE,
+            "smoother_update",
+            sim_fs,
+            target_fs,
+            [],
+            [],
+            [],
         )
 
     @classmethod
     def case_init(cls, sim_fs, mask=None):
-        if mask == None:
-            mask = []
-        return cls(
-            run_type=EnkfRunType.CASE_INIT_ONLY,
-            init_mode=EnkfInitModeEnum.INIT_FORCE,
-            mask=mask,
-            sim_fs=sim_fs,
-            target_fs=None,
-            paths=[],
-            jobnames=[],
-        )
+        mask = [] if mask is None else mask
+        return cls(EnkfRunType.INIT_ONLY, "init_only", sim_fs, None, mask, [], [])
```

Comment:
Is it possible to have `EnkfRunType` not be an Enum? It would be nice to have this open to extension. Ie.:

```
class CaseInitRunContext(RunContext):
    ...
```

Reply:
Probably, think we can do a lot here, I was just working on some RunArg stuff, and looked like we could simplify a lot. Maybe keep it as is for this PR, but create an (or multiple) issue if we merge?

Reply:
Perhaps, it might be really simple and in that case I suggest gunning for it straight away.

Reply:
Removed EnkfRunType, so not relevant anymore.